client/client.go

```diff
@@ -473,6 +473,22 @@ func (w *Wrapper) GetNodePools(clusterID string, p *AuxiliaryParams) (*nodepools
 	return response, nil
 }
 
+// ModifyNodePool modifies a node pool.
+func (w *Wrapper) ModifyNodePool(clusterID, nodePoolID string, modifyNodePoolRequest *models.V5ModifyNodePoolRequest, p *AuxiliaryParams) (*nodepools.ModifyNodePoolOK, error) {
+	params := nodepools.NewModifyNodePoolParams().WithClusterID(clusterID).WithNodepoolID(nodePoolID).WithBody(modifyNodePoolRequest)
+	err := setParamsWithAuthorization(p, w, params)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	response, err := w.gsclient.Nodepools.ModifyNodePool(params, nil)
+	if err != nil {
+		return nil, clienterror.New(err)
```

Comment:
I just noticed `clienterror.New(err)`. What is that? We should do the microerror wrapping here.

Reply:
Let me put that on my refactoring list. This may affect many places, adapting which could make this PR huge.